src/apps/frontend/pages/authentication/otp/otp-form-hook.ts

```diff
@@ -0,0 +1,65 @@
+import { useFormik } from 'formik';
+import * as Yup from 'yup';
+
+import { useAuthContext } from '../../../contexts';
+import { AsyncError } from '../../../types';
+
+interface OTPFormProps {
+  onError: (err: AsyncError) => void;
+  onResendSuccess: () => void;
+  onSuccess: () => void;
+}
+const useOTPForm = ({
+  onError, onResendSuccess, onSuccess,
+}: OTPFormProps) => {
+  const {
+    isVerifyOTPLoading, verifyOTPError, verifyOTPResult, sendOTP, verifyOTP,
+  } = useAuthContext();
+
+  const searchParams = new URLSearchParams(window.location.search);
+  const phoneNumber = searchParams.get('ph');
+  const code = searchParams.get('code');
+
+  const formik = useFormik({
+    initialValues: {
+      otp: Array(4).fill(''),
```

Comment:
`4` is a magic number.

I think you want to store this in a constant (OTP_LENGTH) or something like that and want to ensure that the rest of the code responds dynamically to that. For example, if in future we decide that our OTP lengths are six, we only need to change this variable and rest of the code should be updated accordingly and display 6 boxes etc.